packages/gatsby/src/redux/reducers/jobsv2.js

```diff
@@ -0,0 +1,38 @@
+module.exports = (
+  state = { active: new Set(), stale: new Map(), done: new Map() },
+  action
+) => {
+  switch (action.type) {
+    case `CREATE_JOB_V2`: {
+      const { job, plugin } = action.payload
+
+      state.stale.set(job.contentDigest, {
+        job,
+        plugin,
+      })
+      state.active.add(job.contentDigest)
+
+      return state
+    }
+
+    case `END_JOB_V2`: {
+      const { job, result } = action.payload
+      state.stale.delete(job.contentDigest)
+      state.active.delete(job.contentDigest)
+      // inputPaths is used to make sure the job is still necessary
+      state.done.set(job.contentDigest, { result, inputPaths: job.inputPaths })
```

Comment:
Isn't this a potential memory leak? Unless the `REMOVE_STALE_JOB` is somehow guarnateed to be called, but that kind of makes the whole `.done` map redundant.

Reply:
this should be called every time `gatsby build` or `gatsby develop` is called inside our bootstrap function. 

https://github.com/gatsbyjs/gatsby/pull/19858/files/10b856b4e39713f6d73e658b5961398cfec202a8#diff-8c8b888e741f1c31e97d4bf05894f50dR146-R148

let me rename the function to removeStaleJobs, it says more than getStaleJobs